src/mbgl/annotation/annotation_manager.hpp

```diff
@@ -48,6 +54,9 @@ class AnnotationManager : private util::noncopyable {
     ShapeAnnotationImpl::Map shapeAnnotations;
     std::vector<std::string> obsoleteShapeAnnotationLayers;
     std::set<AnnotationTileMonitor*> monitors;
+
+    SpriteStore spriteStore;
```

Comment:
They are specific to annotations, so I'd rather have them owned by `AnnotationManager`.

Your analysis of the issue sounds correct; however I would rather fix it by using `std::unique_ptr<AnnotationManager>` in `MapData` and resetting that.